### PR TITLE
Fixes robots

### LINKED ISSUE #58368: Robotics unable to create useful bots.
Reporting client version: 513.1538

## Testmerges:
- [Makes the GAS system only update the icon when it has changed](https://www.github.com/tgstation/tgstation/pull/58337)

## Reproduction:
Be a Roboticist and create any little robot, be sad that it refuses to patrol, clean, heal, or detain.
This also affects any robots created other than roundstart, even admin created bots.

### PR DESCRIPTION
**Pull request by Shroopy**
## About The Pull Request
Fixes #58368
Fixes the problem of robots (things like cleanbots, floorbots, NOT borgs) not being useful. Specifically, this fixes their pathing so that they ignore the first step of their pathing, which is the tile that they start on. They used to get stuck trying to move nowhere, and now they properly move.

## Why It's Good For The Game
Functioning robots are fun :)

## Changelog
:cl:
fix: robots now work properly
/:cl:

### CODE REVIEW

**Comment by exymian** [comment]
Thank you you lovely man

**Comment by Shroopy** [comment]
> Thank you you lovely man

Woman, but thank you c:

**Comment by carshalash** [comment]
> > Thank you you lovely man
> 
> Woman, but thank you c:

Thank you, christ Sophia.

File: code/modules/mob/living/simple_animal/bot/bot.dm [hunk 1]
```diff
@@ -526,6 +526,8 @@ Pass a positive integer as an argument to override a bot's default speed.
 		return FALSE
 	var/step_count = move_speed ? move_speed : base_speed //If a value is passed into move_speed, use that instead of the default speed var.
 
+	path.Cut(1,2)
```
Comment (AMonkeyThatCodes) [inline code/modules/mob/living/simple_animal/bot/bot.dm:529]: Add an argument to get_path that runs this before returning the path, this code is currently being copy-pasted everywhere where paths are used.

**Pull request merged**